app/src/main/java/cloudcity/CloudCityConstants.java

```diff
@@ -59,4 +62,10 @@ public class CloudCityConstants {
     public static final String CLOUD_CITY_IPERF3_TEST_TIME_THROTTLING_THRESHOLD = "cloud_city_iperf3_throttling_interval";
     // Iperf3 distance-based throttling shared pref key
     public static final String CLOUD_CITY_IPERF3_TEST_DISTANCE_THROTTLING_THRESHOLD = "cloud_city_iperf3_throttling_distance";
+
+    /**
+     * Whether {@link CellUtil#remapCellClassTypeIntoInteger(CellInformation)} should allow null cells to be
+     * remapped into 0 or not
+     */
+    public static final boolean ALLOW_TESTING_WITHOUT_SIM = false;
```

Comment:
this new `ALLOW_TESTING_WITHOUT_SIM` has nothing to do with the environment, but the actual user of the app. Only certain special builds will have this setting turned on, whether this PR will get merged-in at all is debatable and it might just get scratched/deleted.

But it has nothing to do with the build type (debug/release) or the server environment (staging/demo/production), but the specific user of the app. As such, it will not be automated, at least not for now.